webapp/src/components/systemConsole/teamFilter/teamFilter.tsx

```diff
@@ -32,55 +32,84 @@ const TEAM_FILTER_FILTER_TYPE_OPTIONS: RadioSetting[] = [
+            // getTeams always returns Team[] when includeTotalCount param is set to false
+            const teams = await Client4.getTeams(0, 20, false) as Team[];
+            const options = teams.map((team): DropdownOption => ({
+                value: team.id,
+                label: team.display_name,
+                raw: team,
+            }));
+            setInitialOptions(options);
+        };
 
-            const teamsByID: {[key: string]: Team} = {};
-            const options = teams.
-                filter((team) => team.delete_at === 0).
-                map((team): DropdownOption => {
-                    teamsByID[team.id] = team;
-
-                    return {
-                        value: team.id,
-                        label: team.display_name,
-                        raw: team,
-                    };
-                });
-            setAllTeamsOptions(options);
-
-            const savedSetting = config.PluginSettings.Plugins['com.mattermost.user-survey']?.systemconsolesetting?.TeamFilter;
-            let initialOptions: DropdownOption[] = [];
-            if (savedSetting?.filteredTeamIDs) {
-                initialOptions = savedSetting.filteredTeamIDs.map((teamId) => {
-                    const team = teamsByID[teamId];
-                    return {
-                        label: team?.display_name || `Archived Team: ${teamId}`,
-                        value: teamId,
-                        raw: team,
-                    };
-                });
+        task();
+    }, []);
+
+    const getTeamsByIds = useCallback(async (teamIds: string[]): Promise<Team[]> => {
+        const teams: Team[] = [];
+        const getTeamPromises: Array<Promise<void>> = [];
+
+        const fetchTeam = async (teamId: string) => {
+            const team = await Client4.getTeam(teamId);
+            teams.push(team);
+            return Promise.resolve();
+        };
+
+        teamIds.forEach((teamId) => {
+            getTeamPromises.push(fetchTeam(teamId));
+        });
+
+        await Promise.all(getTeamPromises);
+
+        return teams;
+    }, []);
+
+    // For fetching the teams which are selected so their names can be displayed
+    useEffect(() => {
+        const savedSetting = config.PluginSettings.Plugins['com.mattermost.user-survey']?.systemconsolesetting?.TeamFilter;
+        const initialFilterTypeValue = savedSetting?.filterType || 'everyone';
+        setTeamFilterType(initialFilterTypeValue);
+
+        const task = async () => {
+            if (!savedSetting?.filteredTeamIDs) {
+                return;
             }
 
-            setSelectedTeams(initialOptions);
+            const teamsByID: {[key: string]: Team} = {};
 
-            const initialFilterTypeValue = savedSetting?.filterType || 'everyone';
-            setTeamFilterType(initialFilterTypeValue);
+            // fetch selected teams
+            const teams = await getTeamsByIds(savedSetting.filteredTeamIDs);
```

Comment:
I know this can make many API calls all together, but this would be once, in a very in-frequently accused part of the app.